guides/release/testing/testing-components.md

```diff
@@ -276,7 +276,19 @@ module('Integration | Component | comment-form', function(hooks) {
 
 In cases where components have dependencies on Ember services,
 it is possible to stub these dependencies for rendering tests.
-You stub Ember services by using the built-in `register()` function to register your stub service in place of the default.
+You stub non-Ember services by using the built-in `register()` function to register your stub service in place of the default.
+
+<div class="cta">
+  <div class="cta-note">
+    <div class="cta-note-body">
+      <div class="cta-note-heading">Zoey says...</div>
+      <div class="cta-note-message">
+        Not all services can or should be stubbed. Certain services like router should never be stubbed. If an integration or unit test relies on routing then that behavior would be better tested as an acceptance test. The behavior of internal services like router may change over time and you should not attempt to override their behavior.
```

Comment:
```suggestion
If you are thinking of stubbing the router service in a unit or integration test, consider writing an acceptance test instead. Acceptance tests let you navigate through many routes in your app, and the router does not need to be stubbed in them. If you choose to stub the router, you will need to stub multiple methods.
```

How is this? It's not as strong as "never do this" but hopefully we can steer people in the right direction - either with a clue about multiple methods or the suggestion to write acceptance tests.

Reply:
Much better 👍